components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/graphQL/GraphQLAPIHandler.java

```diff
@@ -0,0 +1,296 @@
+import org.apache.synapse.transport.passthru.util.RelayUtils;
+import org.wso2.carbon.apimgt.gateway.handlers.Utils;
+import org.wso2.carbon.apimgt.gateway.handlers.security.APISecurityConstants;
+import graphql.language.Definition;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.OperationDefinition;
+import graphql.language.OperationDefinition.Operation;
+import graphql.language.Selection;
+import graphql.parser.Parser;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import graphql.schema.idl.UnExecutableSchemaGenerator;
+import graphql.validation.ValidationError;
+import graphql.validation.Validator;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.net.URLDecoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.xml.stream.XMLStreamException;
+
+public class GraphQLAPIHandler extends AbstractHandler {
+
+    private static final String QUERY_PATH_STRING = "/?query=";
+    private static final String REST_SUB_REQUEST_PATH = "REST_SUB_REQUEST_PATH";
+    private static final String API_TYPE = "API_TYPE";
+    private static final String GRAPHQL_API = "GRAPHQL";
+    private static final String UNICODE_TRANSFORMATION_FORMAT = "UTF-8";
+    private static final String INVALID_QUERY = "INVALID QUERY";
+    private static final String SCOPE_ROLE_MAPPING = "ScopeRoleMapping";
+    private static final String SCOPE_OPERATION_MAPPING = "ScopeOperationMapping";
+    private static final String GRAPHQL_IDENTIFIER = "_graphQL";
+    private static final String CLASS_NAME_AND_METHOD = "_GraphQLAPIHandler_handleRequest";
+    private static final Log log = LogFactory.getLog(GraphQLAPIHandler.class);
+    private static GraphQLSchema schema = null;
+    private static Validator validator;
+    private String apiUUID;
+
+    public GraphQLAPIHandler() {
+
+        validator = new Validator();
+    }
+
+    public String getApiUUID() {
+
+        return apiUUID;
+    }
+
+    public void setApiUUID(String apiUUID) {
+
+        this.apiUUID = apiUUID;
+    }
+
+    public boolean handleRequest(MessageContext messageContext) {
+        try {
+            String payload;
+            String operationList = "";
+            Parser parser = new Parser();
+
+            ArrayList<String> operationArray = new ArrayList<>();
+
+            org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
+                    getAxis2MessageContext();
+            String requestPath = ((Axis2MessageContext) messageContext).getProperties().
+                    get(REST_SUB_REQUEST_PATH).toString();
+            if (!requestPath.isEmpty()) {
```

Comment:
Dont we need null check here?

Reply:
Fixed with 0838c21cdd5bb849901b60536393fdd037c427bd